### ISSUE TITLE
There are a number of duplicate simplistic fake IReactorTCP implementations spread throughout the test suite

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3990]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#3990|
|Type|defect|
|Created|2009-08-28 18:41:47Z|
|Branch|https://github.com/twisted/twisted/tree/one-fake-reactor-3990|

These implementations should be unified into a single implementation.  Hopefully that implementation can then grow into a more real thing, offering more of the expected behavior of `IReactorTCP` and perhaps even some of the other reactor interfaces.

(Someday maybe it can even move out of the test suite and be a real thing.)

<details><summary>Searchable metadata</summary>

```
trac-id__3990 3990
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_one_fake_reactor_3990 branches/one-fake-reactor-3990
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1251484907000000 1251484907000000
changetime__1251728231000000 1251728231000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [27255]) Branching to 'one-fake-reactor-3990'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun removed owner|
|-|-|

I consolidated things a bit.

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_3"></a>|@jonathanj set owner to @exarkun|
|-|-|

Looks good, go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [27260]) Merge one-fake-reactor-3990

Author: exarkun
Reviewer: jonathanj
Fixes: [#3990](https://github.com/twisted/twisted/issues/3990)

Merge the several partial implementations of `IReactorTCP` in various test
modules into a single implementation in `twisted.test.proto_helpers`.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
